src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -324,22 +330,18 @@ private boolean checkAndAddNewLogEvent(AtomicInteger totalBytesRead,
      * @param totalBytesRead        The total bytes read till now.
      * @param attemptLogInformation The attempt information containing the log file information.
      * @param data                  The log line read from the file.
-     * @return whether or not the maximum message size has reached or not.
+     * @return no of bytes successfully added to the log attempt for this line.
      * @implNote We need to add extra bytes size for every input message as well as the timestamp byte size alongwith
      *     the log line data size to get the exact size of the input log events.
      */
-    private boolean addNewLogEvent(AtomicInteger totalBytesRead, CloudWatchAttemptLogInformation attemptLogInformation,
+    private AtomicInteger addNewLogEvent(AtomicInteger totalBytesRead, CloudWatchAttemptLogInformation attemptLogInformation,
                                    String data, Instant timestamp) {
         int dataSize = data.getBytes(StandardCharsets.UTF_8).length;
-        // Total bytes equal the number of bytes of the data plus 8 bytes for the timestamp since its a long
-        // and there is an overhead for each log event on the cloud watch side which needs to be added.
-        if (totalBytesRead.get() + dataSize + TIMESTAMP_BYTES + EVENT_STORAGE_OVERHEAD > MAX_BATCH_SIZE) {
-            return true;
-        }
+        AtomicInteger currBytesAdded = new AtomicInteger();
 
         // Cloudwatch does not allow uploading data older than 14 days.
         if (timestamp.isBefore(Instant.now(clock).minus(14, ChronoUnit.DAYS))) {
-            return false;
+            return currBytesAdded;
```

Comment:
do we skip over the event higher up when the added count is `0`? Because that's what needs to happen in this case.

Reply:
I think so, does line 256 address this concern?

Reply:
probably, yes. I believe it is tested, so if the tests are passing we should be OK.

Definitely add a test for your new requirement.

Reply:
not tested yet, just want to get an early opinion while I test and implement tests